common/src/main/java/me/lucko/luckperms/common/calculator/LuckyPerms.java

```diff
@@ -0,0 +1,145 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * The LuckPerms Luckiness Calculator (tm)
+ */
+public final class LuckyPerms {
+
+    /** A sequence of the first 1000 lucky numbers. */
+    private static final Set<Integer> LUCKY_NUMBERS = generateLuckyNumbers(1000);
+    /** Is it an unlucky day? */
+    private static final boolean UNLUCKY_DAY;
+    /** The unlucky number. */
+    private static final int UNLUCKY_NUMBER = 666;
+    /** The lucky number */
+    private static final int LUCKY_NUMBER;
+
+    static {
+        LocalDate date = LocalDate.now();
+        UNLUCKY_DAY = date.getDayOfWeek() == DayOfWeek.FRIDAY && date.getDayOfMonth() == 13;
+
+        if (Locale.getDefault().getLanguage().equals(Locale.CHINESE.getLanguage())) {
+            // https://en.wikipedia.org/wiki/Chinese_numerology#Eight
+            LUCKY_NUMBER = 8;
+        } else {
+            // https://en.wikipedia.org/wiki/7
+            LUCKY_NUMBER = 7;
+        }
+    }
+
+    /** A ~magical~ counter used when determining luckiness. */
+    private final AtomicInteger counter;
+
+    public LuckyPerms() {
+        this.counter = new AtomicInteger();
+    }
+
+    /**
+     * Uses highly advanced, patented LuckPerms(tm) calculations to determine
+     * the current luckiness measure.
+     *
+     * <p>Returns {@link Tristate#TRUE} if lucky, {@link Tristate#FALSE} if unlucky,
+     * and {@link Tristate#UNDEFINED} if unable to determine.</p>
+     *
+     * @return the calculated luckiness measure
+     */
+    public Tristate calculateLuckiness() {
+        int i = this.counter.incrementAndGet();
+
+        // is the counter divisible by the unlucky number? oh no.. that's not a good sign!
+        if (i % UNLUCKY_NUMBER == 0) {
+            return Tristate.FALSE;
+        }
+
+        // got lucky on a dice roll & the counter is in the lucky numbers set? must be lucky!
+        if (isEven(rollDice()) && LUCKY_NUMBERS.contains(i % 1000)) {
+            return Tristate.TRUE;
+        }
+
+        // is it Friday 13th? I sure hope not.
+        if (UNLUCKY_DAY) {
+            return Tristate.FALSE;
+        }
+
+        // is the current system time divisible by the lucky number? sounds promising?!
+        if ((System.currentTimeMillis() / 1000) % LUCKY_NUMBER == 0) {
+            return Tristate.TRUE;
+        }
+
+        // well, this is embarrassing.
+        return Tristate.UNDEFINED;
+    }
+
+    // Roll a 6-sided fair dice.
+    private static int rollDice() {
+        return ThreadLocalRandom.current().nextInt(6) + 1;
+    }
+
+    // Is the number even?
+    private static boolean isEven(int i) {
+        return i % 2 == 0;
+    }
+
+    // Generates 'lucky numbers'
+    // see: https://en.wikipedia.org/wiki/Lucky_number
+    private static Set<Integer> generateLuckyNumbers(int n) {
```

Comment:
I believe we should have a native function in C/C++ with JNI bindings to optimize for performance as we all know Java is not the fastest of languages, **we want these lucky numbers out *fast.***

Yes, the overhead produced by the transition from Java <-> native code is an unfortunate side effect, but worth the operation if we make it generate *even more* lucky numbers, enough to make up for such performance loss.

Reply:
Salutations @Fefo6644 

As the magnanimous director of scrutiny here at the LuckPerms conglomerate, I'd have to hazard a guess that yes, this is possible and likely more performant with large numbers of lucky numbers.

As such, I would like to congratulate you on your prescient advice regarding the swap, as it is an excellent suggestion. However, could there perhaps be some JavaScript package, as @codingJWilliams suggested earlier, which might have similar performance but be native, as Java and JavaScript are native to each other?

As such, perhaps your suggestion or recommendation could be modified to instead use one or many JavaScipt-Node packages, that may have the upsides but not downsides. 

As such, while your suggestion was excellent and appears well-meaning, scrutiny has determined that you could have done better.

With gratitude,

Larry
Magnanimous Director of Scrutiny with the LuckPerms Conglomerate.

Reply:
Hello @Laarryy,

Consider it considered.

Fefo.